pkg/controller/controller.go

```diff
@@ -111,6 +111,18 @@ func (c *controller) serviceBindingCallback(e k8swatch.Event) error {
 			return err
 		}
 		glog.Infof("Created Service Binding: %s\n%v\n", sb.Name, created)
+	} else if e.Type == k8swatch.Deleted {
+		// if we've already processed the deletion event, ignore
+		if sb.DeletionTimestamp != nil {
+			return nil
+		}
+
+		if err := c.handler.DeleteServiceBinding(&sb); err != nil {
+			return err
+		}
+	} else if e.Type == k8swatch.Modified && sb.DeletionTimestamp != nil {
```

Comment:
@MHBauer modifications when the timestamp is nil mean that someone messed with the TPR after they created it. I don't know what the chances are that they'll do so, though.

For completeness' sake, note that the timestamp is our internal flag to indicate that we're doing "finalization" see https://github.com/kubernetes-incubator/service-catalog/pull/300#discussion_r99189317 for a more in-depth discussion on this finalization mechanism. 

we had to decide what those modifications mean in this PR, and we decided that we don't support them, so those do indeed fall into the `WarningF` below.